### PR TITLE
feat: extend Connection class to accept external provider

### PR DESCRIPTION
**Pull request by iamcrazycoder**
This PR extends the `Connection` class to accept external provider. The changes include updating the `ConnectionConfig` (2nd argument of the constructor) and adding a new prop in it: `sendAsync`. 

Parent PR: https://github.com/ExodusMovement/solana-web3.js/pull/4
Asana ticket: https://app.asana.com/0/0/1201512430798460/f

### CODE REVIEW

**Comment by iamcrazycoder** [comment]
Tested the changes. Our provider passed in `ConnectionConfig.sendAsync` is working as expected by routing the requests.

IMG_0C1F4C7CB5FA-1

Logs:

```
received from dapp '{"jsonrpc":"2.0","method":"ExodusWeb3Provider_mainnet_getEpochInfo","params":[],"id":781}'
sent to dapp '{"jsonrpc":"2.0","id":778,"result":{"absoluteSlot":118206875,"blockHeight":106695953,"epoch":273,"slotIndex":270875,"slotsInEpoch":432000,"transactionCount":53867836150}}'
received from dapp '{"jsonrpc":"2.0","method":"ExodusWeb3Provider_mainnet_getEpochInfo","params":[],"id":782}'
sent to dapp '{"jsonrpc":"2.0","id":779,"result":{"absoluteSlot":118206877,"blockHeight":106695955,"epoch":273,"slotIndex":270877,"slotsInEpoch":432000,"transactionCount":53867838258}}'
```

**Comment by iamcrazycoder** [comment]
Reminder: update newly released package version in https://github.com/ExodusMovement/web-dapps/pull/14

File: src/connection.ts [hunk 1]
```diff
@@ -1982,6 +1993,11 @@ export type FetchMiddleware = (
   fetch: (modifiedUrl: string, modifiedOptions: any) => void,
 ) => void;
 
+export type ExternalProvider = (request: {
+  method: string;
+  params?: Array<any>;
+}) => Promise<void>;
```
Comment (kklash) [inline src/connection.ts:1999]: doesn't the provider return a JSON RPC response object (not void)?

**Review by kklash — COMMENTED** [review]
Looks great! one suggestion

File: src/connection.ts (hunk 1 above)
Comment (iamcrazycoder) [reply src/connection.ts:1999]: Good catch @kklash. I will verify that in the morning.

**Comment by iamcrazycoder** [comment]
As explained in [this TDD](https://github.com/ExodusMovement/engineering-handbook/pull/348), the `Connection` and `Transaction` classes are loosely coupled in Solana and integration of our provider in Connection is class is no longer needed. 

We have concluded that only requests related to transactions (e.g., `signTransaction` / `signMessage`) need to be routed via our provider and validated by the exodus-api-validation layer.

We can do that by injecting our SDK into dApps on load in exodus-mobile and calling the provider from there.

[Asana task ](https://app.asana.com/0/1201717857544522/1201746035633074/f)
[Slack thread](https://exodusio.slack.com/archives/CM8KFHJ0K/p1643407875705769)

cc/ @kklash @bulgakovk

**Pull request closed without merging**